domain/secretbackend/service/service.go

```diff
@@ -700,9 +700,53 @@ func (s *Service) RotateBackendToken(ctx context.Context, backendID string) erro
 	return errors.Trace(err)
 }
 
-// SetModelSecretBackend sets the secret backend for the given model UUID.
+// GetModelSecretBackend returns the secret backend name for the given model UUID, returning an error
+// satisfying [modelerrors.NotFound] if the model provided does not exist.
+func (s *Service) GetModelSecretBackend(ctx context.Context, modelUUID coremodel.UUID) (string, error) {
+	backendName, modelType, err := s.st.GetModelSecretBackend(ctx, modelUUID)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	switch modelType {
+	case coremodel.IAAS:
+		if backendName == provider.Internal {
+			backendName = provider.Auto
+		}
+	case coremodel.CAAS:
```

Comment:
Small nit but both of these case statements can be pushed together and use `&& backendName == kubernetes.BackendName`

Reply:
No, we cannot.
For caas model, `kubernetes` -> `auto` but `internal` no change.
For iaas model, `internal` -> `auto` but `kubernetes` no change.